pkg/agent/plugin/nodeattestor/aws/iid.go

```diff
@@ -20,32 +21,43 @@ import (
 )
 
 const (
-	iidPluginName               = "aws_iid"
-	defaultIdentityDocumentUrl  = "http://169.254.169.254/latest/dynamic/instance-identity/document"
-	defaultIdentitySignatureUrl = "http://169.254.169.254/latest/dynamic/instance-identity/signature"
+	defaultIdentityDocumentHost  = "169.254.169.254"
+	defaultIdentityDocumentPath  = "/latest/dynamic/instance-identity/document"
+	defaultIdentitySignatureHost = "169.254.169.254"
+	defaultIdentitySignaturePath = "/latest/dynamic/instance-identity/signature"
 )
 
+// IIDAttestorConfig configures a IIDAttestorPlugin.
 type IIDAttestorConfig struct {
-	IdentityDocumentUrl  string `hcl:"identity_document_url"`
-	IdentitySignatureUrl string `hcl:"identity_signature_url"`
+	IdentityDocumentHost  string `hcl:"identity_document_host"`
```

Comment:
I can add them back in and have URL preferred over Host and Path for now, though that will start getting messy. I do agree that it would be better for stability though.
As far as the comment on why this is being done:
--it better enables unit testing
--in case of a change to amazon's API or host info (whether breaking or in providing different/new info), users of this can update their config without needing to wait for the codebase to be updated

Reply:
I'm not sure I understand the win for unit-testing in breaking out the host and path portions. How is that materially different from the URL? I agree there is arguably less validation since you don't have to check for bad scheme values or unexpected query or fragment values, but other than that...

In the same vein, if the Amazon API endpoint shifted to a different host or path, that could be just as easily updated via a URL config. The URL approach is even a little more flexible, since it allows operators to pick up a change in procotol (e.g. endpoint becomes https).

If the configuration started as Host and Path, I wouldn't be looking to change them until needed. In the same vein I'm a little wary of changing from URL to Host/Path without a clear need.